lib/modules/versioning/devbox/index.spec.ts

```diff
@@ -0,0 +1,78 @@
+import devbox from '.';
+
+describe('modules/versioning/devbox/index', () => {
+  it.each`
+    version           | expected
+    ${'1'}            | ${false}
+    ${'01'}           | ${false}
+    ${'1.01'}         | ${false}
+    ${'1.1'}          | ${false}
+    ${'1.3.0'}        | ${true}
+    ${'2.1.20'}       | ${true}
+    ${'v1.4'}         | ${false}
+    ${'V0.5'}         | ${false}
+    ${'3.5.0'}        | ${true}
+    ${'4.2.21.Final'} | ${false}
+    ${'1234'}         | ${false}
+    ${'foo'}          | ${false}
+    ${'latest'}       | ${false}
+    ${''}             | ${false}
+    ${'3.5.0-beta.3'} | ${false}
+    ${'*'}            | ${false}
+    ${'x'}            | ${false}
+    ${'X'}            | ${false}
+    ${'~1.2.3'}       | ${false}
+    ${'>1.2.3'}       | ${false}
+    ${'^1.2.3'}       | ${false}
+    ${'1.2.3-foo'}    | ${false}
+    ${'1.2.3foo'}     | ${false}
+  `('isVersion("$version") === $expected', ({ version, expected }) => {
+    expect(!!devbox.isVersion(version)).toBe(expected);
+  });
+
+  it.each`
+    version           | isValid
+    ${'1'}            | ${true}
+    ${'01'}           | ${false}
+    ${'1.01'}         | ${false}
+    ${'1.1'}          | ${true}
+    ${'1.3.0'}        | ${true}
+    ${'2.1.20'}       | ${true}
+    ${'v1.4'}         | ${false}
+    ${'V0.5'}         | ${false}
+    ${'3.5.0'}        | ${true}
+    ${'4.2.21.Final'} | ${false}
+    ${'1234'}         | ${true}
+    ${'foo'}          | ${false}
+    ${'latest'}       | ${true}
+    ${''}             | ${false}
+    ${'3.5.0-beta.3'} | ${false}
+    ${'*'}            | ${false}
+    ${'x'}            | ${false}
+    ${'X'}            | ${false}
+    ${'~1.2.3'}       | ${false}
+    ${'>1.2.3'}       | ${false}
+    ${'^1.2.3'}       | ${false}
+    ${'1.2.3-foo'}    | ${false}
+    ${'1.2.3foo'}     | ${false}
+  `('isValid("$version") === $isValid', ({ version, isValid }) => {
+    expect(!!devbox.isValid(version)).toBe(isValid);
+  });
+
+  it.each`
+    version    | range       | expected
+    ${'1'}     | ${'1'}      | ${false}
+    ${'1'}     | ${'0'}      | ${false}
+    ${'1.2.3'} | ${'1'}      | ${true}
+    ${'1.2'}   | ${'1'}      | ${false}
+    ${'1.0.0'} | ${'1'}      | ${true}
+    ${'1.2.0'} | ${'1.2'}    | ${true}
+    ${'1.2.3'} | ${'1.2'}    | ${true}
+    ${'0'}     | ${'latest'} | ${false}
```

Comment:
```suggestion
    ${'1.2.3'} | ${'latest'} | ${true}
    ${'0'}     | ${'latest'} | ${false}
```

Should latest match against any version like this?

Reply:
I don't think we can really know what latest matches without making a request to nixhub.
In the case where they have put latest I think it would be safest to not try to update it with renovate and let the user call `devbox update "package"` to do the upgrade.
Or I could check for latest in the upgrade step and call that from renovate.

Reply:
Is there any potential issues from the equals/matches functions always returning false when comparing against "latest"?

Reply:
If latest is treated the same as * then Renovate will take care of matching it to the highest version. Is there is a lock file for devbox then this is something you'd want. If there's no lock file then it doesn't matter too much.

Reply:
Yeah there is a lockfile, how do I get renovate to treat "latest" like "*"?

Reply:
Make sure that matches(x, 'latest') always returns true

Reply:
ok got it, I'll implement that now

Reply:
That is now updated